text/3840-compat-math-identifiers.md

```diff
@@ -0,0 +1,205 @@
+- Feature Name: `compat_math_identifiers`
+- Start Date: 2025-07-16
+- RFC PR: [rust-lang/rfcs#3840](https://github.com/rust-lang/rfcs/pull/3840)
+- Rust Issue: [TODO rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Rust already supports a wide range of unicode characters in identifiers - for example `α`, `номер`, `عدد`, `数`, `संख्या` are all valid Rust identifiers.
+This RFC extends the set of Unicode character which can be used in identifiers with [`ID_Compat_Math_Start`](https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5B%3AID_Compat_Math_Start%3DYes%3A%5D&g=&i=idtype) and [`ID_Compat_Math_Continue`](https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5B%3AID_Compat_Math_Continue%3DYes%3A%5D&g=&i=idtype), most notable: `∇`, `∂`, `∞`, subscripts `⁰¹²³⁴⁵⁶⁷⁸⁹⁺⁻⁼⁽⁾` and superscripts `₀₁₂₃₄₅₆₇₈₉₊₋₌₍₎`.
+This can be a boon to implementers of scientific concepts as they can write for example `let ∇E₁₂ = 0.5;`.
+
+# Motivation
+[motivation]: #motivation
+
+Programming languages have historically focused on the quite narrow set of ASCII characters, however developers from other cultures or specialized problem spaces can benefit from using characters which are native to their culture or domain.
+The vast body of scientific literature uses a variety of characters to express concepts from physics, mathematics, biology, robotics and many others.
+Symbols often appearing in equations are Roman letters like `x`, Greek letters like `θ`, and differentiation operators like `∂` and `∇`.
+Variables are often adorned with subscripts like `x₁₂` or superscripts like `x⁺` or `x⁽²⁾`.
+Having these symbols available as Rust identifiers could simplify the implementation of these concepts and stay closer to a reference publication, thus reducing confusing and implementation errors.
```

Comment:
Grammar: typically it's "reducing confusion and implementation errors" or "reducing confusing errors and implementation errors" 

```suggestion
Having these symbols available as Rust identifiers could simplify the implementation of these concepts and stay closer to a reference publication, thus reducing confusion, and implementation errors.
```